_clients/data-prepper/data-prepper-reference.md

```diff
@@ -151,12 +151,39 @@ acmCertificateArn | Conditionally | String | Represents the ACM certificate ARN.
 
 ### string_converter
 
-Converts strings to uppercase or lowercase. Mostly useful as an example if you want to develop your own prepper.
+Converts string to uppercase or lowercase. Mostly useful as an example if you want to develop your own processor.
 
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
 upper_case | No | Boolean | Whether to convert to uppercase (`true`) or lowercase (`false`).
 
+### aggregate
+
+Groups events together based on the keys provided and performs a action on each group.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+identification_keys | Yes | List | A unordered list by which to group Events. Events with the same values for these keys are put into the same group. If an Event does not contain one of the `identification_keys`, then the value of that key is considered to be equal to `null`. At least one identification_key is required.
+action | Yes | Object | The action to be performed for each group. One of the Aggregate Actions must be provided. `remove_duplicates` and `put_all` are existing actions.
+group_duration | No | String | The amount of time that a group should exist before it is concluded automatically. Supports ISO_8601 notation Strings ("PT20.345S", "PT15M", etc.) as well as simple notation Strings for seconds (`"60s"`) and milliseconds (`"1500ms"`). Default value is `180s`.
+
+### date
+
+Adds a default timestamp to the event or parses timestamp fields, and converts it to ISO 8601 format which can be used as event timestamp.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+match | Conditionally | List | List of `key` and `patterns` where patterns is a list. List can of only of size 1. There is no default value. `from_time_received` and `match` are mutually exclusive options. Use multiple date processor instance if both options should be used.
```

Comment:
"List can of only of size 1." -> "The list of patterns must have exactly one pattern in it." (or something similar rewording)